public/js/widget-manager.js

```diff
@@ -44,8 +44,9 @@ define(['jupyter-js-widgets'], function(Widgets) {
      */
     WidgetManager.prototype.display_view = function(msg, view, options) {
         var widgetInfo = this._pendingExecutions[msg.parent_header.msg_id];
-        // no longer need to track
-        delete this._pendingExecutions[msg.parent_header.msg_id];
+
+        // keep tracking the message ID in case more than one widget will reside
+        // in the widget area
```

Comment:
Apparently we can't, ever. There's no indication when some message might come from the kernel instantiating a new widget. Could be delayed and out of order with other messages. Could be dynamic based on the notebook logic.

I was probably in the wrong to try to prematurely optimize here. Since the nodes never go away in the DOM (yet) anyway, there's really no harm in keeping them around. And it sure looks like we have to anyway.